File: Winforms Example 9 - Table Properties Dialog/frmCellProperties.vb
Imports System.Text

Public Class frmCellProperties

    Public Cancelled As Boolean = True

    Private Sub btnOK_Click(sender As Object, e As EventArgs) Handles btnOK.Click
        Cancelled = False
        Me.Hide()
    End Sub

    Private Sub btnCancel_Click(sender As Object, e As EventArgs) Handles btnCancel.Click
        Cancelled = True
        Me.Hide()
    End Sub

    Private Function GetColor() As Color

        Dim oColorDialog As New ColorDialog
        Dim oDialogResult As DialogResult

        oDialogResult = oColorDialog.ShowDialog(Me)

        If oDialogResult = DialogResult.OK Then
            Return oColorDialog.Color
        Else
            Return Nothing
        End If

    End Function

    Private Sub btnCellBorderColor_Click(sender As Object, e As EventArgs) Handles btnCellBorderColor.Click

        Dim oColor As Color = GetColor()

        If Not oColor = Nothing Then

            Dim sb As StringBuilder = New StringBuilder()
            sb.AppendFormat("#")
            sb.Append(BitConverter.ToString(New Byte() {oColor.R}))
            sb.Append(BitConverter.ToString(New Byte() {oColor.G}))
            sb.Append(BitConverter.ToString(New Byte() {oColor.B}))

            txtCellBorderColor.Text = sb.ToString
        End If

    End Sub

    Private Sub btnCellBackgroundColor_Click(sender As Object, e As EventArgs) Handles btnCellBackgroundColor.Click

        Dim oColor As Color = GetColor()

        If Not oColor = Nothing Then

            Dim sb As StringBuilder = New StringBuilder()
            sb.AppendFormat("#")
            sb.Append(BitConverter.ToString(New Byte() {oColor.R}))
            sb.Append(BitConverter.ToString(New Byte() {oColor.G}))
            sb.Append(BitConverter.ToString(New Byte() {oColor.B}))

            txtCellBackgroundColor.Text = sb.ToString
        End If

    End Sub

End Class

File: Winforms Example 09 - Table Properties Dialog/frmCellProperties.cs
using System;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace Winforms_Example_09___Table_Properties_Dialog_CS
{
    public partial class frmCellProperties : Form
    {
        public bool Cancelled = true;

        public frmCellProperties()
        {
            InitializeComponent();
        }

        private void btnOK_Click(object sender, EventArgs e)
        {
            Cancelled = false;
            this.Hide();
        }

        private void btnCancel_Click(object sender, EventArgs e)
        {
            Cancelled = true;
            this.Hide();
        }

        private Color GetColor()
        {
            ColorDialog oColorDialog = new ColorDialog();
            DialogResult oDialogResult;

            oDialogResult = oColorDialog.ShowDialog(this);

            if (oDialogResult == DialogResult.OK)
                return oColorDialog.Color;
            else
                return Color.Empty;
        }

        private void btnCellBorderColor_Click(object sender, EventArgs e)
        {
            Color oColor = GetColor();

            if (oColor != null/* TODO Change to default(_) if this is not a reference type */ )
            {
                StringBuilder sb = new StringBuilder();
                sb.AppendFormat("#");
                sb.Append(BitConverter.ToString(new byte[] { oColor.R }));
                sb.Append(BitConverter.ToString(new byte[] { oColor.G }));
                sb.Append(BitConverter.ToString(new byte[] { oColor.B }));

                txtCellBorderColor.Text = sb.ToString();
            }
        }

        private void btnCellBackgroundColor_Click(object sender, EventArgs e)
        {
            Color oColor = GetColor();

            if (oColor != null/* TODO Change to default(_) if this is not a reference type */ )
            {
                StringBuilder sb = new StringBuilder();
                sb.AppendFormat("#");
                sb.Append(BitConverter.ToString(new byte[] { oColor.R }));
                sb.Append(BitConverter.ToString(new byte[] { oColor.G }));
                sb.Append(BitConverter.ToString(new byte[] { oColor.B }));

                txtCellBackgroundColor.Text = sb.ToString();
            }
        }
    }
}
